modules/tinymce/src/plugins/table/main/ts/queries/TabContext.ts

```diff
@@ -60,17 +61,18 @@ const handle = (event: KeyboardEvent, editor: Editor) => {
     };
 
     const rng = editor.selection.getRng();
-    if (rng.collapsed) {
-      const start = SugarElement.fromDom(rng.startContainer);
-      TableLookup.cell(start, isRoot).each((cell) => {
-        event.preventDefault();
-        const navigation = event.shiftKey ? backward : forward;
-        const rng = navigation(editor, isRoot, cell);
-        rng.each((range) => {
-          editor.selection.setRng(range);
-        });
+    // If navigating backwards, use the start of the ranged selection
+    const container = SugarElement.fromDom(event.shiftKey ? rng.startContainer : rng.endContainer);
+    TableLookup.cell(container, isRoot).each((cell) => {
+      event.preventDefault();
+      // Clear fake ranged selection because our new selection will always be collapsed
+      TableLookup.table(cell, isRoot).each(cellSelection.clear);
```

Comment:
Just wondering, should we only do this when setting the range, as that means it's found a valid navigation location?

Reply:
Discussed in Slack: Moving this will mean the fake selection is present when adding rows (which uses the `mceTableInsertRowAfter` command) resulting in multiple rows. So we'll leave this where it is.